apollo-core/src/main/java/com/ctrip/framework/apollo/core/ApolloClientSystemConsts.java

```diff
@@ -161,4 +161,32 @@ public class ApolloClientSystemConsts {
    * enable apollo overrideSystemProperties
    */
   public static final String APOLLO_OVERRIDE_SYSTEM_PROPERTIES = "apollo.override-system-properties";
+
+  /**
+   * apollo client monitor enabled
+   */
+  public static final String APOLLO_CLIENT_MONITOR_ENABLED = "apollo.client.monitor.enabled";
+
+  /**
+   * apollo client monitor metrics event pool size
+   */
+  public static final String APOLLO_CLIENT_MONITOR_METRICS_EVENT_POOL_SIZE = "apollo.client.monitor.metrics-event-pool-size";
```

Comment:
event pool seems a litter dangerous, and if exists concurrent event usage, that will let data chaos.

Reply:
I will refer to the online implementation to try to write a thread-safe and good-performing object pool.

Reply:
I don’t really think it will have thread safety issues with data loss. It will most likely have thread safety issues with the reset function of the pool.